test/functionalTest/cases/0000_deprecated_checkings/legacy_wgs84.test

```diff
@@ -46,7 +46,7 @@ payload='{
         {
           "name": "location",
           "type": "string",
-          "value": "WSG84"
```

Comment:
This .test case is designed to test the support for "WGS84" literal. It was used in old versions of Orion before we note it was wrong, but we needed to keep support for not breaking backward compatibility.

What would happen if WGS84 is used in this .test? It breaks?

Reply:
Yes, because the Context Broker now only holds location metadata as "custom", so it is rendered as it was stored during POST /v2/entities. No hardcoded location metadata additions, which caused duplication, wrongly inclusions and inconsistency among v1/v2 GET/POSTs.

Reply:
Let's pospone this particular .test until the ones described at https://github.com/telefonicaid/fiware-orion/pull/3346#issuecomment-433969358 get completed.

Reply:
This test was postponed while the other .test gets completed. Now is the moment to resume discussing :)

As far as I understand, this .test doesn't needs a change. I'll try to explin graphically. At creation stage:

![imagen](https://user-images.githubusercontent.com/1534240/48834626-910cd680-ed7e-11e8-9c1e-ab3e792ef8dc.png)

What to you think?

Reply:
The creation stage matches with the current behavior, thus it's ok. It creates the location field because it finds a location metadata in NGSIv1, which is stored as is ("WSG84").

In the DB we have the following:
```
{
    "_id" : {
        "id" : "Madrid",
        "type" : "City",
        "servicePath" : "/"
    },
    "attrNames" : [ 
        "location"
    ],
    "attrs" : {
        "location" : {
            "type" : "coords",
            "creDate" : 1542813795,
            "modDate" : 1542813795,
            "value" : "40.418889, -3.691944",
            "md" : {
                "location" : {
                    "type" : "string",
                    "value" : "WSG84"
                }
            },
            "mdNames" : [ 
                "location"
            ]
        }
    },
    "creDate" : 1542813795,
    "modDate" : 1542813795,
    "location" : {
        "attrName" : "location",
        "coords" : {
            "type" : "Point",
            "coordinates" : [ 
                -3.691944, 
                40.418889
            ]
        }
    },
    "lastCorrelator" : "5e1fd5c4-eda1-11e8-8e61-0242ac110002"
}
```

The test fails in the retrieve operation. As a location metadata (even if WSG84) is present, it is render as it is (WSG84) and not as WGS84. We should add a dedicated check.
 So,  I modified the following:
```
/* The flag below indicates that a location metadata with WGS84 was found during iteration.
        * It needs to the NGSIV1 check below, in order to add it if the flag is false */
        if (md->name == NGSI_MD_LOCATION && md->stringValue == LOCATION_WGS84)
        {
          noLocationMetadata = false;
        }
```
In this way, either if the location was not present at all or present but with any wrong value (including WSG84), the right WGS84 metadata is added. The problem is that now it returns both:
```
{
    "contextResponses": [
        {
            "contextElement": {
                "type": "City",
                "isPattern": "false",
                "id": "Madrid",
                "attributes": [
                    {
                        "name": "location",
                        "type": "coords",
                        "value": "40.418889, -3.691944",
                        "metadatas": [
                            {
                                "name": "location",
                                "type": "string",
                                "value": "WSG84"
                            },
                            {
                                "name": "location",
                                "type": "string",
                                "value": "WGS84"
                            }
                        ]
                    }
                ]
            },
            "statusCode": {
                "code": "200",
                "reasonPhrase": "OK"
            }
        }
    ]
}
```
I guess we should add some logic to check explicitly that a WSG84 metadata but not a WGS84 was added or can we keep the response reported above?

Reply:
We should remove WSG84 from metadataVector and add the right WGS84 ? Maybe this logic should be implemented in the metadata value rendering logic (by replacing there directly the WSG84 value with WGS84), in the latter case, we would avoid to deal with metadataVector pop (WSG84) and then push (WGS84)

Reply:
I solved with:
```
  /* The flag below indicates that a location metadata with WGS84 was found during iteration.
        *  It needs to the NGSIV1 check below, in order to add it if the flag is false
        *  In addition, adjust old wrong WSG84 metadata value with WGS84 */
        if (md->name == NGSI_MD_LOCATION)
        {
          noLocationMetadata = false;

          if (md->valueType == orion::ValueTypeString && md->stringValue == LOCATION_WGS84_LEGACY)
          {
            md->stringValue = LOCATION_WGS84;
          }
        }

        caP->metadataVector.push_back(md);
      }
    }

    if (apiVersion == V1)
    {
      /* Setting location metadata (if location attr found
       *  and the location metadata was not present or was present but with old wrong WSG84 value) */
      if ((locAttr == ca.name) && (ca.type != GEO_POINT) && noLocationMetadata)
      {
        /* Note that if attribute type is geo:point then the user is using the "new way"
         * of locating entities in NGSIv1, thus location metadata is not rendered */
        Metadata* md = new Metadata(NGSI_MD_LOCATION, "string", LOCATION_WGS84);
        caP->metadataVector.push_back(md);
      }
    }
```

Fixed in 6bc0cc4